packages/mco/components/modals/app-manage-policies/app-manage-policies-modal.tsx

```diff
@@ -78,13 +79,15 @@ export const ModalContextViewer: React.FC<ModalContextViewerProps> = ({
 };
 
 export const AppManagePoliciesModal: React.FC<AppManagePoliciesModalProps> = ({
-  applicaitonInfo,
+  applicationInfo: appInfo,
   matchingPolicies,
   loaded,
   loadError,
   isOpen,
   close,
 }) => {
+  // Perform deep comparison to prevent unnecessary re-renders
+  const applicationInfo = useDeepCompareMemoize(appInfo);
```

Comment:
```suggestion
  const applicationInfo = useDeepCompareMemoize(appInfo, true);
```
`appInfo` is a very deep object (contains multiple CRs)... can you try with `true` argument and check if that works ??

Reply:
@SanjalKatiyar does this provide better performance?

Reply:
it's a small object, it wont impact the performance

Reply:
looks like I was wrong: https://www.samanthaming.com/tidbits/33-how-to-compare-2-objects/#json-stringify-vs-lodash-s-isequal-performance

Reply:
for small objects stringfy is better and for large nested objects lodash is better !

Reply:
i will check with true option

Reply:
Based on the comment: [[GitHub link](https://github.com/red-hat-storage/odf-console/pull/1878#discussion_r1969562795)](https://github.com/red-hat-storage/odf-console/pull/1878#discussion_r1969562795), I have moved the memoization one step back. The issue specifically occurs for `appset`, so I have relocated the memoization logic inside `application-set-parser.tsx`.  

In this section ([[GitHub link](https://github.com/red-hat-storage/odf-console/pull/1878/files#diff-ecc0b5ba3f2a4cf0a22eec5c4aed84f598b5114bfb31fcb8acee4ecee28ddf93R105)](https://github.com/red-hat-storage/odf-console/pull/1878/files#diff-ecc0b5ba3f2a4cf0a22eec5c4aed84f598b5114bfb31fcb8acee4ecee28ddf93R105)), the object size is slightly larger, so I have used Lodash for optimization.